rubi/rubi/client.py

```diff
@@ -453,395 +461,605 @@ def _default_event_handler(
         )
 
         if raw_event.client_filter(wallet=self.wallet):
-            pair = self._pairs.get(pair_name)
-
             if isinstance(raw_event, EmitFeeEvent):
-                event = FeeEvent.from_event(pair=pair, event=raw_event)
+                asset = self.network.tokens[raw_event.asset]
+
+                event = FeeEvent.from_event(
+                    pair_name=pair_name, asset=asset, event=raw_event
+                )
             else:
+                base_asset, quote_asset = pair_name.split("/")
+
                 event = OrderEvent.from_event(
-                    pair=pair, event=raw_event, wallet=self.wallet
+                    base_asset=self.network.tokens[base_asset],
+                    quote_asset=self.network.tokens[quote_asset],
+                    event=raw_event,
+                    wallet=self.wallet,
                 )
 
-                self._update_active_limit_orders(order_events=[event])
-
             self.message_queue.put(event)
 
     ######################################################################
     # order methods
     ######################################################################
 
-    def place_market_order(self, transaction: Transaction) -> TransactionReceipt:
```

Comment:
i wonder if we should keep this function and transform the `Transaction` object to the a `NewMarketOrder`? i guess this really continues to boil down to how we want to handle the deprecation of functions throughout the repo